### PR TITLE
tests: more updates on periph_uart

### PR DESCRIPTION
**Pull request by smlng**
follow up on #17

### CODE REVIEW

File: tests/periph_uart/tests/01__periph_uart_base.robot [hunk 1]
```diff
@@ -16,64 +16,73 @@ Variables           test_vars.py
...
     API Call Should Succeed     Uart Write  ${SHORT_TEST_STRING}
     API Result Data Should Contain  ${SHORT_TEST_STRING}
-    Show PHILIP Statistics
+    Log                 PHILIP Stats
 
-Baud Test 9600
-    PHILIP.Setup Uart           baudrate=9600
-    UART Init and Flush Should Succeed      baud=${9600}
+Baudrate 38400 Should Succeed
+    [Documentation]     Verify UART write with baudrate 9600.
```
Comment (MrKevinWeiss) [inline tests/periph_uart/tests/01__periph_uart_base.robot]: oops

File: tests/periph_uart/tests/01__periph_uart_base.robot [hunk 2]
```diff
@@ -16,64 +16,73 @@ Variables           test_vars.py
...
 *** Test Cases ***
-Echo
+Short Echo Should Succeed
+    [Documentation]     Write short string to UART and verify echo reply.
     PHILIP.Setup Uart
     UART Init and Flush Should Succeed
     API Call Should Succeed     Uart Write  ${SHORT_TEST_STRING}
     API Result Data Should Contain  ${SHORT_TEST_STRING}
-    Show PHILIP Statistics
+    Log                 PHILIP Stats
```
Comment (MrKevinWeiss) [inline tests/periph_uart/tests/01__periph_uart_base.robot]: It appears that this just logs the string `PHILIP Stats` and not any values

File: tests/periph_uart/tests/01__periph_uart_base.robot (hunk 1 above)
Comment (smlng) [reply tests/periph_uart/tests/01__periph_uart_base.robot]: oh

File: tests/periph_uart/tests/01__periph_uart_base.robot (hunk 2 above)
Comment (smlng) [reply tests/periph_uart/tests/01__periph_uart_base.robot]: mhm, that's weird

**Comment by smlng** [comment]
note: needs squashing now

**Comment by smlng** [comment]
so **do not** insta merge

**Comment by smlng** [comment]
HIL was happy (apart from the usual fails) so I squashed

**Pull request merged**